### DISCUSSION TITLE
SSO Okta integration without Dex and Argo CD

### DISCUSSION THREAD

**Discussion by dmildh-absci** [argoproj/argo-workflows#13380] [Q&A] [answered]
Hello everyone.  Is is possible to enable SSO using Okta in Argo Workflows without using Argocd's dex implementation? We are using Workflows but not CD and don't want to have to install and configure another dependency to get it to work.

Thank you.

**Comment by agilgur5** [comment, marked as answer]
You can use a standalone Dex instance with the same configuration, it doesn't have to be Argo CD's Dex instance

**Reply by dmildh-absci** [reply]
I was able to get Okta based SSO for Argo CD working all from a single helm install so was just thinking it would be nice to be able to do something similar with the workflows chart.  I guess the assumption for this chart is that you are already using the CD component?

**Reply by agilgur5** [reply]
Dex is not needed to run Workflows, so it's an optional component. It's also not necessarily needed for SSO with Workflows; it depends on the capabilities of your provider. I believe people have configured Okta without Dex but I don't remember exactly.

Also this repo has no Helm Charts. You can certainly check [`argo-helm`](https://github.com/argoproj/argo-helm) and make a feature request for an optional Dex component there if it hasn't been asked for previously (search the issues).
